csrc/codegen.cpp

```diff
@@ -1281,6 +1281,108 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
     }
   }
 
+  void handle(const ArgsortOp* aop) final {
+    NVF_ERROR(isAligned(), "Unaligned argsort not supported");
```

Comment:
QQ: this just means that we need predicate?

Reply:
No, it means we can't have divergent threads. Maybe it should be renamed to something like `maybeDiverged`. We did this as an optimization since we can use `__syncthreads` instead of more generic version of synchronization.